File: csharp/prime-factors/PrimeFactors.cs
public static class PrimeFactors
{
    public static long[] Factors(long number)
    {
        List<long> factors = [];
        long divisor = 2;

        while (number > 1)
        {
            if (number % divisor == 0)
            {
                factors.Add(divisor);
                number /= divisor;
            }
            else
            {
                divisor++;
            }
        }
        return factors.ToArray();
    }
}

File: javascript/prime-factors/prime-factors.js
//
// This is only a SKELETON file for the 'Prime Factors' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const primeFactors = (number) => {
  const factors = [];
  let divisors = 2;

  while (number > 1) {
    if (number % divisors === 0) {
      factors.push(divisors);
      number /= divisors;
    } else {
      divisors++;
    }
  }
  return factors;
};
